### ISSUE TITLE
Distribution 3.11 tracking issue

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#6695]
| Mark | Description                                       |
|------|---------------------------------------------------|
| 🏃   | Work in progress                                  |
| ✋   | Blocked task                                      |
| 💪   | Stretch goal for this iteration                   |
| ⏱️💪   | Status changed to stretch mid-iteration             | 
| ⌛   | Waiting for information from user          |
| 🎁   | Unexpected addition to milestone         |
| 🔴   | Missing issue reference                           |
| 🔵   | More investigation required to remove uncertainty |
| ⚫️   | Under discussion within the team                  |
| ~crossed out~ | No longer expect to achieve in iteration. |

The work we set out to do in 3.10 was: https://github.com/sourcegraph/sourcegraph/issues/6038

## Planned work

- Stephen
  - [x] 3.10.3: add monitoring & dashboard for search requests by status (success, error, timeout, partial timeout) #7053
  - [x] 3.10.3: When search hits unindexed repositories, a poor timeout UX / error is returned #6754
  - [x] Searcher: cannot unmarshal number into Go value of type protocol.NotFoundPayload #6911
  - [x] 🎁🏃Proposal PR: Remove the management console (🦄unicorns and 🌈rainbows here) #7197
   - [x] ⏱️💪(carried over from 3.9) Some repositories stuck in "Empty repository" state #5684
   - [x] 🏃⏱️💪(carried over from 3.9) Better logging around GraphQL API usage #5706
  - [x] 🏃⏱️💪Search pagination: decide better behavior for cloning/missing/timedout/partial repos #6000
  - [ ] ~⌛Single-container docker instances should not randomly crash (issue https://app.hubspot.com/contacts/2762526/ticket/54947439/)~
   - [ ] ~⏱️💪Search pagination: approximateResultCount is certainly wrong #6002~
   - [ ] ~⏱️💪Search pagination: preemptively fetch and cache results for users that are likely to come back soon #6001~
   - [ ] ~⏱️💪Search pagination: potentially update and reason about bucket bounds / divisor #5998~
   - [ ] ~⏱️💪Search pagination: Add support for non-file (text) result types #5997~
   - [ ] ~⏱️💪🔴Push monitoring needed by $CUSTOMER~
- Uwe
  - [ ] 🏃high level metrics for all services
  - [ ] 🏃make alerting first class citizen (builtin in all deploy contexts)
  - [ ] 🏃alert gitserver latency spikes https://github.com/sourcegraph/sourcegraph/issues/6651
  - [ ] 🏃disk space alerts https://github.com/sourcegraph/sourcegraph/issues/5994
  - [ ] 💪(carried over from 3.9) finalize Dashboards/Panels and document them
  - [ ] 💪eliminate ephemeral alerts [RFC 35](https://docs.google.com/document/d/1GXYTBktfRjr3I5vyl-TwCixITtKG0snVXcehRU1dfok/edit#heading=h.9gknqva39m5l)
  - [ ] 💪metric labeling https://github.com/sourcegraph/sourcegraph/issues/6650
  - [ ] 💪alerts for code intelligence latency (gitlab integration)
  - [x] metric labeling https://github.com/sourcegraph/sourcegraph/issues/6618
  - [x] metric labeling https://github.com/sourcegraph/sourcegraph/issues/6599
  - [ ] 💪attribute SameSite for cookies https://github.com/sourcegraph/sourcegraph/issues/6167
  - [ ] ✋management console authentication [RFC 63](https://docs.google.com/document/d/1RkOS4EehLtAXhunTazkjCI4yKi5Hc8eRcHZ_v1Fz_QU/edit?ts=5db76618#)
  - [ ] ✋exempt Grafana from CSRF protection middleware
- Beyang
  - [x] Release captain automation (https://github.com/sourcegraph/sourcegraph/pull/6801, https://github.com/sourcegraph/about/pull/388)
  - [x] 🏃 Updates to release/QA process (https://docs.google.com/document/d/1mbnLHBhEPT2c8ttoxQaQRstivW0JIbKHKtjg-ZRr5WM/edit)
  - [x] 🏃 Digital Ocean installation process (cert error) https://github.com/sourcegraph/sourcegraph/issues/6652
  - [ ] 🏃 Update resource recommendations for single Docker. Also audit k8s resource requirements. Document process for sizing instances (overprovision first, then decrease based on resource utilization metrics) https://github.com/sourcegraph/sourcegraph/issues/6274, https://github.com/sourcegraph/sourcegraph/issues/2019
  - [x] Global settings via env var https://github.com/sourcegraph/sourcegraph/issues/5014
  - [x] https://github.com/sourcegraph/sourcegraph/issues/4695
  - [x] Quick fixes: https://github.com/sourcegraph/sourcegraph/issues/4209, https://github.com/sourcegraph/sourcegraph/issues/2476
  - [ ] 💪 Security (replace with concrete items after reviewing RFC)
  - [ ] 💪 Multiple authz-enabled external services with same host: https://github.com/sourcegraph/sourcegraph/issues/6736
  - [ ] 💪 E2E CI flakiness https://github.com/sourcegraph/sourcegraph/issues/6169, https://github.com/sourcegraph/sourcegraph/issues/5721 (Keegan also working on this, revisit if this is still necessary after we have a separate e2e pipeline.)
  - [ ] GraphQL request timeout https://github.com/sourcegraph/sourcegraph/issues/6878
- Geoffrey
  - [x] ci: move e2e tests to separate buildkite pipeline: https://github.com/sourcegraph/sourcegraph/issues/6840
  - [x] Parallelize Docker build scripts: https://github.com/sourcegraph/sourcegraph/pull/6841
  - [x] ci: use asdf to lock versions of frequently used build tools (go and node): https://github.com/sourcegraph/sourcegraph/issues/6862
  - [x] ci: use buildkit to parallelize separate stages: https://github.com/sourcegraph/sourcegraph/pull/6864
  - [x] ci: run dev/check/all.sh checks in parallel: https://github.com/sourcegraph/sourcegraph/pull/7104
  - [x] ci: refactor foreach-ts-project to run in parallel" https://github.com/sourcegraph/sourcegraph/pull/7107
  - [x] ci: add --mutex network to all yarn installs: https://github.com/sourcegraph/sourcegraph/pull/7131
  - [ ] 🏃: TODO write up issue for caching yarn build artifcats for CI pipeline

## Interrupts

- [x] https://github.com/sourcegraph/sourcegraph/issues/6911 @beyang

**Comment by beyang** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.11 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you
